data_discovery_ai/utils/es_connector.py

```diff
@@ -26,49 +26,98 @@
 def connect_es(config: configparser.ConfigParser) -> Elasticsearch:
     end_point = config["elasticsearch"]["end_point"]
     api_key = config["elasticsearch"]["api_key"]
-
-    client = Elasticsearch(end_point, api_key=api_key)
-    logging.info("Connected to ElasticSearch")
-    return client
+    try:
+        client = Elasticsearch(end_point, api_key=api_key)
+        logging.info("Connected to ElasticSearch")
+        return client
+    except Exception as e:
+        logger.error(e)
 
 
 """
     Search elasticsearch index, convert the json format to dataframe, save the dataframe to a pickle file
     Input:
         client: Elasticsearch. The initialised Elasticsearch client instance
+        index: str. The index name in Elasticsearch. Default as "es-indexer-staging"
+        batch_size: int. The number of documents searches at one query. Please not set a very large number in case to search too deep.
+        sleep_time: int. The number of seconds of sleep time between each query. This should be set with consideration of 'batch_size'. If 'batch_size' is large, set 'sleep_time' to a large number to ensure each query is finished and does not impact the next query.
     Output:
         raw_data: pd.DataFrame. The fetched raw data in a tabular format.
 """
 
 
-def search_es(client: Elasticsearch):
-    index = "es-indexer-staging"
-    start_index = 0
+def search_es(
+    client: Elasticsearch,
+    index: str = "es-indexer-staging",
+    batch_size: int = 100,
+    sleep_time: int = 5,
+):
     dataframes = []
-    batch_size = 100
-    resp = client.search(index=index, body={"query": {"match_all": {}}})
-    total_number = resp["hits"]["total"]["value"]
 
+    # get document count
+    count_resp = client.count(index=index)
+    total_number = count_resp["count"]
```

Comment:
what will happen if `count_resp` is None or `count_resp["count"]` is None?

Reply:
try-except and if condition for validation have been added to capture the fields raised from es response